apiserver/embeddedcli.go

```diff
@@ -70,7 +70,11 @@ func (h *embeddedCLIHandler) ServeHTTP(w http.ResponseWriter, req *http.Request)
 		ticker := time.NewTicker(websocket.PingPeriod)
 		defer ticker.Stop()
 
-		modelUUID := httpcontext.RequestModelUUID(req)
+		modelUUID, valid := httpcontext.RequestModelUUID(req)
```

Comment:
The embeddedcli is the ability to call juju directly in a server instance. I'm not fan of the implementation. Anyway, if the model UUID isn't valid we need to error out. 

> The comment above says this function is for executing commands directly on the controller

This is correct.

> which even suggests the normal case here would not have a modeluuid since the request would be to the controller directly

This is in the context of a modeluuid for the command. Running `juju` and accessing the REPL gives you some idea of what it can do.